symforce/codegen/backends/python/templates/util/util.jinja

```diff
@@ -217,3 +217,27 @@ def {{ function_name_and_args(spec) }}:
         {%- if not loop.last %}, {% endif %}
     {%- endfor -%}
 {% endmacro %}
+
+{# ------------------------------------------------------------------------- #}
+
+{# Macro to flatten an array if it's an ndarray and is a vector. Also, raises
+ # a ValueError if the length is not equal to size and the shape is not that of
+ # a vector.
+ #
+ # Args:
+ #     array (str): The name of an sequence that might be an ndarray we wish to flatten
+ #     size (int): The expected size of the array
+ #}
+{% macro flatten_if_ndarray(array, size) %}
+if isinstance({{ array }}, numpy.ndarray):
+    if {{ array }}.shape in [({{ size }}, 1), (1, {{ size }})]:
+        {{ array }} = {{ array }}.flatten()
+    elif {{ array }}.shape != ({{ size }},):
+        raise ValueError("Expected {{ array }} to be a vector of length {{ size }}; instead had shape {}".format({{ array }}.shape))
+elif len({{ array }}) != {{ size }}:
+    raise ValueError(
+        "Expected {{ array }} to be a sequence of length {{ size }}, was instead length {}.".format(
+            len({{ array }})
+        )
+    )
+{%- endmacro %}
```

Comment:
One thing I couldn't quite decide here was whether I should convert everything to a list inside the macro.

Arguments for converting to a list:
- Using `tolist` converts the weird normal number types to builtin types (`list()` does not do this)
![image](https://user-images.githubusercontent.com/84410250/196313939-1962bf57-b2cb-48a5-8686-9d1b105f30cc.png)
If the builtin types are preferable, then we'd need to convert inside of this macro, as outside of it we're not sure if the array is an `ndarray` or some other type.

Arguments against converting to a list:
- If we are just going to iterate over the array (like we do in the `__init__` methods of the camera cal classes), converting from an iterable to a list is pointless and just performs an unnecessary copy.
- Also slightly complicates things with the type checker (I imagine) as we're changing the type of the array. Would probably need to assign to a new variable).

Reply:
I'm not sure if it's going to matter?  Also confirmed the same thing happens for floats as for ints.  Some unanswered questions:

1) Is there some reason to prefer things be ints? Probably the builtin types are faster?
2) On the other hand it does seem possibly useful to e.g. make one of these objects out of `np.float32`?  But I'm not sure if anyone would actually want to do that or care, especially since all the generated functions would immediately break that.
3) Are other things going to break unexpectedly if you try to do things with a sym class made of numpy scalars?  E.g. what if you try to call some pybind method on one of these?

I'm probably leaning towards calling `tolist`?  Probably not worth investigating any/all of those questions

Reply:
I don't think it matters (now that I've clarified my thoughts by typing out the above), as I don't myself know any reason to prefer one datatype over the other. And I would be very surprised if numpy scalars caused things to be broken, as the conversion works fine (pybind has built in Eigen <--> numpy conversions), and I don't know how anything else could go wrong.

Reply:
To be clear, I've stuck with it as it is (i.e., no `tolist`) simply because I'm kind of leaning against it (just for the reasons I outlined above). If you had a good argument for `tolist`, I could change, but in the absence of that...